### PR TITLE
qemu: correct installation dir for firmware's json

### PR DESCRIPTION
**Pull request by sgn**
Close: #23495

---
@Hoshpak

### CODE REVIEW

**Comment by Hoshpak** [comment]
Am 16.07.20 um 17:18 schrieb Danh Doan:
> Close: #23495
> 
> ---
> @Hoshpak 

I guess it works but I don't like it much. We shouldn't apply yet
another hack just to work around that hook which creates a false
positive in this case. My preferred way to handle this would be to make
it possible to define exceptions to the hook and just install the files
in the proper directory in the first place. I will try to come up with
an alternative PR tonight.

File: srcpkgs/qemu/template [hunk 1]
```diff
@@ -81,6 +81,15 @@ do_install() {
 	make DESTDIR=${DESTDIR} install
 	# qemu-bridge-helper must be setuid for non privileged users.
 	chmod u+s ${DESTDIR}/usr/libexec/qemu-bridge-helper
+	# The firmware JSON files should be searched for in three directories
+	# * /usr/share/qemu/firmware
+	# * /etc/qemu/firmware
+	# * $XDG_CONFIG_HOME/qemu/firmware
+	# See: https://git.qemu.org/?p=qemu.git;a=blob;f=docs/interop/firmware.json;h=240f565397ae0e754e85976f2d4b5e3873ae8211;hb=HEAD#l311
+	vmkdir usr/share/qemu
+	mv ${DESTDIR}/usr/lib/qemu/firmware ${DESTDIR}/usr/share/qemu/
```
Comment (ahesford) [inline srcpkgs/qemu/template:90]: If the ultimate destination is `usr/share/qemu` cant `--datadir=/usr/lib` be removed or replaced with `--datadir=/usr/share` when calling `configure`?

Comment (sgn) [reply srcpkgs/qemu/template:90]: No,
Right now, the firmware is installed into `/usr/lib/qemu`, i.e. `/usr/lib/qemu/multiboot.bin` those files are sources of warnings in pkg-lint-elf-in-usr-share.

The json files in `/usr/lib/qemu/firmware` must be moved, let's see what Hoshpak propose.

**Comment by sgn** [comment]
On 2020-07-16 10:13:38-0700, Helmut Pozimski <notifications@github.com> wrote:
> I guess it works but I don't like it much. We shouldn't apply yet
> another hack just to work around that hook which creates a false
> positive in this case. My preferred way to handle this would be to make
> it possible to define exceptions to the hook and just install the files
> in the proper directory in the first place. I will try to come up with
> an alternative PR tonight.

I don't like it either.
Just too lazy to fix the pkglint.

-- 
Danh

**Pull request closed without merging**